streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -131,6 +166,21 @@ private void determineIsQueryable(final String queryableStoreName) {
     }
 
     @Override
+    public <VR> KTable<K, VR> aggregate(final Initializer<VR> initializer,
```

Comment:
Should we have an overload without `Materialized`, since we have these for `WindowedKStream` and `SessionedKStream`?

Reply:
We already do: `aggregate(Initializer, Aggregator, Serde)` this is the one that has only the required params

Reply:
I meant for `aggregate(Initializer, Aggregator)`, i.e. without having `Serde` also and use defaults.

Reply:
do you mean having `aggregate(Initializer, Aggregator)` in addition to the one with the `Serde`? If that is the case i think we'd need both as i imagine it would be pretty common to change the type during an aggregation and it would be overkill to have to provide a `Materialized` just to change the agg `Serde`. 
Further, the `Serde` param is nullable if you want to use the defaults, so do we really want to add another overload?

Reply:
Actually, I meant to replace `aggregate(Initializer, Aggregator, Serde)` with `aggregate(Initializer, Aggregator)`. I.e. to remove the third parameter since if users do want to override the serde, they should be using the `Initializer, Aggregator, Materialized` overload, right?

Reply:
I think we should leave it as is. Having to materialise a state store just for a serde seems like overkill to me.

Reply:
@guozhangwang having thought about it a bit more, i think we should just have the 2 overloads, i.e, `aggregate(Initializer, Aggregator)` and `aggregate(Initializer, Aggregator, Materialized)` as it is consistent with everything else